tests/test_dynamodb.py

```diff
@@ -0,0 +1,233 @@
+        ],
+        AttributeDefinitions=[
+            {"AttributeName": "doi", "AttributeType": "S"},
+        ],
+    )
+    dynamodb_class.client.put_item(
+        TableName="test_dois",
+        Item={
+            "doi": {"S": "111.1/1111"},
+            "status": {"S": "Failed, will retry"},
+            "attempts": {"S": "10"},
+        },
+    )
+    validation_status = dynamodb_class.retry_attempts_exceeded(
+        "test_dois", "111.1/1111", "10"
+    )
+    assert validation_status is True
+
+
+@mock_dynamodb2
+def test_dynamodb_update_doi_item_attempts_in_database(dynamodb_class):
+    dynamodb_class.client.create_table(
+        TableName="test_dois",
+        KeySchema=[
+            {"AttributeName": "doi", "KeyType": "HASH"},
+        ],
+        AttributeDefinitions=[
+            {"AttributeName": "doi", "AttributeType": "S"},
+        ],
+    )
+    dynamodb_class.client.put_item(
+        TableName="test_dois",
+        Item={
+            "doi": {"S": "111.1/1111"},
+            "status": {"S": "Failed, will retry"},
+            "attempts": {"S": "1"},
+        },
+    )
+    existing_item = dynamodb_class.client.get_item(
+        TableName="test_dois",
+        Key={"doi": {"S": "111.1/1111"}},
+    )
+    assert existing_item["Item"] == {
+        "attempts": {"S": "1"},
+        "doi": {"S": "111.1/1111"},
+        "status": {"S": "Failed, will retry"},
+    }
+    update_response = dynamodb_class.update_doi_item_attempts_in_database(
+        "test_dois", "111.1/1111"
+    )
+    assert update_response["ResponseMetadata"]["HTTPStatusCode"] == 200
+    updated_item = dynamodb_class.client.get_item(
+        TableName="test_dois",
+        Key={"doi": {"S": "111.1/1111"}},
+    )
+    assert updated_item["Item"] == {
+        "attempts": {"S": "2"},
+        "doi": {"S": "111.1/1111"},
+        "status": {"S": "Failed, will retry"},
+    }
+
+
+@mock_dynamodb2
+def test_dynamodb_update_doi_item_status_in_database_invalid_enum(
+    caplog, dynamodb_class
+):
+    with caplog.at_level(logging.DEBUG):
+        dynamodb_class.client.create_table(
+            TableName="test_dois",
+            KeySchema=[
+                {"AttributeName": "doi", "KeyType": "HASH"},
+            ],
+            AttributeDefinitions=[
+                {"AttributeName": "doi", "AttributeType": "S"},
+            ],
+        )
+        dynamodb_class.client.put_item(
+            TableName="test_dois",
+            Item={
+                "doi": {"S": "111.1/1111"},
+                "status": {"S": "Failed, will retry"},
+                "attempts": {"S": "1"},
+            },
+        )
+        existing_item = dynamodb_class.client.get_item(
+            TableName="test_dois",
+            Key={"doi": {"S": "111.1/1111"}},
+        )
+        assert existing_item["Item"] == {
+            "attempts": {"S": "1"},
+            "doi": {"S": "111.1/1111"},
+            "status": {"S": "Failed, will retry"},
```

Comment:
should be changed to something like Status.Failed, after the enum change has been made.